unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeHQImageHandler.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using DCL;
+using DCL.Helpers.NFT;
+using UnityEngine;
+
+public class NFTShapeHQImageConfig
+{
+    public string contentType;
+    public NFTInfo nftInfo;
+    public NFTShapeConfig nftConfig;
+    public NFTShapeLoaderController controller;
+    public AssetPromise_Texture previewTexture;
+}
+
+public class NFTShapeHQImageHandler : IDisposable
+{
+    NFTShapeHQImageConfig config;
+    AssetPromise_Texture hqTexture;
+
+    bool isPlayerNear;
+    bool isCameraInFront;
+    bool isPlayerLooking;
+
+    Camera camera;
+    Vector2 screenCenter;
+    RaycastHit raycastHitInfo;
+
+    static public NFTShapeHQImageHandler Create(NFTShapeHQImageConfig config)
+    {
+        if (config.contentType == "image/gif")
+            return null;
+
+        return new NFTShapeHQImageHandler(config);
+    }
+
+    public void Dispose()
+    {
+        CommonScriptableObjects.playerUnityPosition.OnChange -= OnPlayerPositionChanged;
+
+        if (hqTexture != null)
+        {
+            ForgetHQTexture();
+        }
+    }
+
+    public void Update()
+    {
+        if (config.controller.collider == null)
+            return;
+
+        if (!isPlayerNear)
+            return;
+
+        isCameraInFront = Vector3.Dot(config.controller.transform.forward, config.controller.transform.position - camera.transform.position)
+            > config.nftConfig.hqImgInFrontDotProdMinValue;
+
+        if (config.nftConfig.verbose)
```

Comment:
What are the benefits of having this `verbose` flag in the config instead of a const? I dont think this will change at runtime, does it?

Reply:
I did change it in runtime while debugging :P
but you are right, I will make it a constant